glambie/processing/process_regional_results.py

```diff
@@ -266,48 +355,23 @@ def _run_region_timeseries_one_source(
     log.info("Using the following annual datasets for %s, %s : %s",
              data_group.long_name, region.long_name, str([d.user_group for d in data_catalogue_annual.datasets]))
 
-    # convert to annual trends
-    data_catalogue_annual = convert_datasets_to_annual_trends(data_catalogue_annual, year_type=year_type,
-                                                              method_to_correct_seasonally=method_to_correct_seasonally,
-                                                              seasonal_calibration_dataset=seasonal_calibration_dataset)
-    # convert to mwe
-    data_catalogue_annual = convert_datasets_to_unit_mwe(data_catalogue_annual)
-
-    # calculate combined annual timeseries
-    # first need to recombine timeseries in cases where we split them due to gaps
-    # reason for this is that we want to remove trends over a common period
-    if len(split_dataset_names_annual) > 0:
-        data_catalogue_annual = recombine_split_timeseries_in_catalogue(
-            data_catalogue_annual, split_dataset_names_annual)
-    # then average timeseries from annual catalogue, removing the trends
-    annual_combined, catalogue_annual_anomalies = data_catalogue_annual.average_timeseries_in_catalogue(
-        remove_trend=True, out_data_group=data_group)
+    annual_combined, data_catalogue_annual_homogenized, catalogue_annual_anomalies = \
+        _run_region_variability_for_one_source(
+            data_catalogue_annual=data_catalogue_annual, seasonal_calibration_dataset=seasonal_calibration_dataset,
+            year_type=year_type, method_to_correct_seasonally=method_to_correct_seasonally,
+            data_group=data_group, dataset_names_where_split_at_gap=split_dataset_names_annual)
 
     # 2) LONGTERM TRENDS
     log.info("Using the following trend datasets for %s, %s : %s",
              data_group.long_name, region.long_name, str([d.user_group for d in data_catalogue_trends.datasets]))
-    log.info("Recalibrating with longterm trends within data group and region...")
-    # get catalogue with longerm datasets in same unit as calibration dataset (mwe)
-    data_catalogue_trends = convert_datasets_to_longterm_trends_in_unit_mwe(
-        data_catalogue_trends, year_type=year_type,
-        seasonal_calibration_dataset=seasonal_calibration_dataset,
-        method_to_extract_trends=method_to_extract_trends,
-        method_to_correct_seasonally=method_to_correct_seasonally,
-        output_trend_date_range=min_max_time_window_for_longterm_trends)
 
-    # now treat case where trends are outside annual combined timeseries
-    annual_combined_full_ext = extend_annual_timeseries_if_outside_trends_period(
-        annual_timeseries=annual_combined,
-        data_catalogue_trends=data_catalogue_trends,
-        timeseries_for_extension=annual_backup_dataset.convert_timeseries_to_annual_trends(year_type=year_type))
-
-    # recalibrate
-    catalogue_calibrated_series = calibrate_timeseries_with_trends_catalogue(
```

Comment:
I can almost see the "committee" nature of the project in the repo structure; it would be good to take some time at the beginning of the next phase of GlaMBIE to refactor this after the ATBD is complete for this phase, I'm sure there's a lot of stuff here that looks more complicated than it is purely because of the order in which it was implemented.